include/nanobind/eigen/dense.h

```diff
@@ -241,6 +249,107 @@ struct type_caster<Eigen::Ref<T, Options, StrideType>> {
     operator Ref() { return Ref(caster.operator Map()); }
 };
 
+/// Caster for Eigen::SparseMatrix<T>
+template <typename T> struct type_caster<T, enable_if_t<is_eigen_sparse_v<T>>> {
+    using Scalar = typename T::Scalar;
+    using StorageIndex = std::remove_reference_t<decltype(*std::declval<T>().outerIndexPtr())>;
+    using Index = typename T::Index;
+    static constexpr bool rowMajor = T::IsRowMajor;
+
+    using ScalarCaster = make_caster<tensor<Scalar>>;
+    using StorageIndexCaster = make_caster<tensor<StorageIndex>>;
+    using IndexCaster = make_caster<Index>;
+
+    NB_TYPE_CASTER(T, const_name<rowMajor>("scipy.sparse.csr_matrix[",
+					   "scipy.sparse.csc_matrix[")
+		   + make_caster<Scalar>::Name + const_name("]"));
+
+    ScalarCaster scalar_caster;
+    StorageIndexCaster storageindex_caster;
+    IndexCaster index_caster;
+
+    bool from_python(handle src, uint8_t flags, cleanup_list *cleanup) noexcept {
+	if (!src) {
+	    return false;
+	}
+
+	auto obj = borrow<object>(src);
+	object sparse_module = module_::import_("scipy.sparse");
+	object matrix_type = sparse_module.attr(rowMajor ? "csr_matrix" : "csc_matrix");
+
+	if (!obj.type().is(matrix_type)) {
+            try {
```

Comment:
expand this `try` block so that it also includes the `module::import_`, etc. bits. These can raise exceptions, and the type caster is *not* allowed to raise.